Phase01Project1/src/InvertedIndex.java

```diff
@@ -0,0 +1,48 @@
+import Documents.Document;
+import Normalizer.TokenNormalization;
+import Tokenize.Tokenizer;
+
+import java.util.*;
+
+public class InvertedIndex {
+
+    private ArrayList<Document> documents;
+    private Map<String, ArrayList<Document>> wordDocuments;
+    private TokenNormalization tokenNormalization;
+    private SearchQuery searchQuery;
+
+    public InvertedIndex(SearchQuery searchQuery, ArrayList<Document> documents){
+        wordDocuments =  new HashMap<>();
+        this.searchQuery = searchQuery;
+        this.tokenNormalization = searchQuery.tokenNormalization;
+        this.documents = documents;
+    }
+
+    public void fillWordDocument(String delimiter){
+        Tokenizer tokenizer = new Tokenizer();
+        for(Document document : documents){
+            for(String documentWord : tokenizer.tokenize(document.getText(), delimiter)){
+                documentWord = tokenNormalization.makeNormalizeAndStem(documentWord);
+                if(!wordDocuments.containsKey(documentWord))
+                    wordDocuments.put(documentWord, new ArrayList<>());
+                if(!wordDocuments.get(documentWord).contains(document))
+                    wordDocuments.get(documentWord).add(document);
+            }
+        }
```

Comment:
normalizer should be an interface and the object of tokenizer must be passed in constructor of class. why initializing a tokenizer and it is hard coded?